Drag.py

```diff
@@ -0,0 +1,189 @@
+    elif(choice == 'z'):
+        force_msg.wrench.force.z = -appliedLinearForce
+    elif(choice == 'yw'):
+        force_msg.wrench.torque.z = appliedTorque
+    elif(choice == 'rl'):
+        force_msg.wrench.torque.x = appliedTorque
+    elif(choice == 'p'):
+        force_msg.wrench.torque.y = appliedTorque
+
+    pub.publish(force_msg)
+
+    while not (rospy.is_shutdown()):
+        '''
+        On each iteration of while loop: velocity is gained at two points in time (deltat = 2 seconds)
+        the velocity is compared by taking the absolute value of the difference of the two velocities
+        if the abs of the difference of the two velocities is smaller than small delta value,
+        the two velocities are assumed to be approximately equal. If two velocities taken at different
+        time intervals are equal, the velocity is assumed to be maximized, reaching terminal velocity.
+        '''
+        rospy.Subscriber("/odom", Odometry, get_velocity, choice)
+        velocity1 = Velocity
+        rospy.sleep(2)
+        velocity2 = Velocity
+        Compare_Velocities = abs(velocity1 - velocity2)
+        if Compare_Velocities < delta:
+            Max_Velocity = velocity2
+            Calculate_Drag(choice)  # Once velocity is max, calculate drag using max velocity
+            break  # When the velocities are 'equal', the loop breaks.
+
+    # Once drag is calculated, apply wrench with force/torque of zero to slow sub in that axis
+    if(choice == 'x'):
+        force_msg.wrench.force.x = 0
+    elif(choice == 'y'):
+        force_msg.wrench.force.y = 0
+    elif(choice == 'z'):
+        force_msg.wrench.force.z = 0
+    elif(choice == 'yw'):
+        force_msg.wrench.torque.z = 0
+    elif(choice == 'rl'):
+        force_msg.wrench.torque.x = 0
+    elif(choice == 'p'):
+        force_msg.wrench.torque.y = 0
+
+    pub.publish(force_msg)
+
+    while(Velocity > delta and not rospy.is_shutdown()):
+        # While loop stops program from proceeding until the sub has basically stopped movement in that direction
+        rospy.sleep(2)
+        continue
+
+
+if __name__ == '__main__':
+    try:
+        # Buoyancy is calculated
+        Find_Bouyancy('z')
+        # Drag Coeffcients are calculated in each axis
+        Apply_Force('z')
+        Apply_Force('y')
+        Apply_Force('x')
+        Apply_Force('yw')
+        Apply_Force('rl')
+        Apply_Force('p')
+        # Drag coefficients are written to file, do with them what you wish.
+        file_object = open("DragCoefficients", 'w')
+        file_object.write("Drag Coefficients:")
+        file_object.write("\nLinear X: " + str(Linear_Drag_X))
+        file_object.write("\nLinear Y: " + str(Linear_Drag_Y))
+        file_object.write("\nLinear Z: " + str(Linear_Drag_Z))
+        file_object.write("\nRoll: " + str(Roll_Drag))
+        file_object.write("\nPitch: " + str(Pitch_Drag))
+        file_object.write("\nYaw: " + str(Yaw_Drag))
+        file_object.close()
```

Comment:
Ditto'ing Kevin's comment, definitely consider dumping the data into a format that's computer readable, a yaml like what Kevin mentions would be great. It would make it much easier to dump things into Python to extract data, graph, etc.